### PR TITLE
Enforce java.nio.charset.StandardCharsets against guava Charsets

### PR DESCRIPTION
**Pull request by Abacn**
StandardCharsets is introduced in JRE 7, and guava Charset is now a delegate to java.nio Charsets, e.g.

https://github.com/google/guava/blob/8bab177c14b6c3faf82d6959a9c774b0d15e7414/guava/src/com/google/common/base/Charsets.java#L67

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/java/build-tools/src/main/resources/beam/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -119,6 +119,14 @@ page at http://checkstyle.sourceforge.net/config.html -->
       <property name="message" value="You are using raw guava, please use vendored guava classes."/>
     </module>
 
+    <!-- Forbid guava Charset. -->
```
Comment (Abacn) [inline sdks/java/build-tools/src/main/resources/beam/checkstyle/checkstyle.xml:122]: tested that the enforcement is effective

without source code changes, build fails with

```
> Task :sdks:java:core:checkstyleMain
[ant:checkstyle] [ERROR] /Users/yathu/beam/sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/providers/LoggingTransformProvider.java:39: You are using guava Charsets, please use Java builtin StandardCharsets. [ForbidGuavaCharset]
...


```

**Comment by Abacn** [comment]
testInvalidRecordReceived (org.apache.beam.sdk.io.gcp.spanner.changestreams.SpannerChangeStreamErrorTest) failed unrelated flaky test. It passed on the first commit of this PR

**Comment by Abacn** [comment]
assign set of reviewers

**Comment by Abacn** [comment]
This has been causing internal check violation (ref: go/standardcharsets)

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**